src/Microsoft.Azure.WebJobs.Host/Executors/JobHostContext.cs

```diff
@@ -78,7 +78,7 @@ public IAsyncCollector<FunctionInstanceLogEntry> FastLogger
 
         public void Dispose()
         {
-            if (_disposed)
```

Comment:
I think this hasn't been a big issue noticed by users because generally when their Console app host is shutting down, they don't care about proper disposal since the process is going down anyways.

However in Azure Functions scenarios, we're starting/stopping hosts in memory and proper cleanup is very important.